crates/sui-types/src/type_input.rs

```diff
@@ -148,44 +148,9 @@ impl TypeInput {
         }
     }
 
-    /// Convert to a `TypeTag` consuming `self`. This can fail if this value includes invalid
-    /// identifiers.
-    pub fn into_type_tag(self) -> Result<TypeTag> {
-        use TypeInput as I;
-        use TypeTag as T;
-        Ok(match self {
-            I::Bool => T::Bool,
-            I::U8 => T::U8,
-            I::U16 => T::U16,
-            I::U32 => T::U32,
-            I::U64 => T::U64,
-            I::U128 => T::U128,
-            I::U256 => T::U256,
-            I::Address => T::Address,
-            I::Signer => T::Signer,
-            I::Vector(t) => T::Vector(Box::new(t.into_type_tag()?)),
-            I::Struct(s) => {
-                let StructInput {
-                    address,
-                    module,
-                    name,
-                    type_params,
-                } = *s;
-                let type_params = type_params
-                    .into_iter()
-                    .map(|t| t.into_type_tag())
-                    .collect::<Result<_>>()?;
-                T::Struct(Box::new(StructTag {
-                    address,
-                    module: Identifier::new(module)?,
-                    name: Identifier::new(name)?,
-                    type_params,
-                }))
-            }
-        })
-    }
-
     /// Conversion to a `TypeTag`, which can fail if this value includes invalid identifiers.
+    /// NB: This function should _not_ be used in the adapter or on the "write" side of transaction
+    /// processing.
     pub fn as_type_tag(&self) -> Result<TypeTag> {
```

Comment:
Nit I think this function is named wrong, and it should be `to_type_tag` since it clones? `as_` is for & -> &